gpflow/expectations.py

```diff
@@ -230,7 +230,15 @@ def expectation(p, obj1, obj2=None, nghp=None):
     try:
         return _expectation(p, obj1, feat1, obj2, feat2, nghp=nghp)
     except NotImplementedError as e:
-        print(str(e))
+
+        def custom_warning_format(msg, *args, **kwargs):
```

Comment:
I'm not sure about this lines. Do you still need this?

Reply:
Yes, I would keep it. Otherwise the old msg was warning which was not informative at all. These two lines are the only way that I know of for format warning msg.

Also, unlike other print statements we eliminated, this is not a debug logger, we really want the user to know that specific expectation is solved by quadrature, if that is expected behaviour  then the user can choose to ignore the warning. 

I am not a fan of printing on stdout and logger is not appropriate for this case, as we are explicitly catching an error.

Reply:
a lot of things have changed since you created this PR, we added logger and converted some warning to switchable logs

Reply:
I know, but this is still valid case, as you can see original statement is still print.

Reply:
point is that it is not really a scope of this task, It is global problem which is solvable by changing logger level or turning off warning messages

Reply:
btw, `expectations.py` don't use warnings anymore, only logger